types/clownface/lib/Clownface.d.ts

```diff
@@ -0,0 +1,75 @@
+import { Dataset, Term, BlankNode, NamedNode, Literal, Quad_Graph } from 'rdf-js';
+
+type TermOrClownface = Clownface | Term;
+type TermOrLiteral = TermOrClownface | string | number | boolean;
+
+type AddCallback<D extends Dataset, X extends Term> = (added: Clownface<D, X>) => void;
+type SingleOrArray<T> = T | T[];
+
+type SingleOrArrayOfTerms = SingleOrArray<TermOrClownface>;
+type SingleOrArrayOfTermsOrLiterals = SingleOrArray<TermOrLiteral>;
+
+interface NodeOptions {
+    type?: 'BlankNode' | 'Literal' | 'NamedNode';
+    datatype?: Term | { toString(): string };
+    language?: string;
+}
+
+interface ClownfaceOptions<D extends Dataset> {
+    dataset?: D;
+    graph?: Quad_Graph;
+    term?: Term | Term[];
+    value?: string | string[];
+    _context?: any;
+}
+
+interface Clownface<D extends Dataset = Dataset, T extends Term = Term> {
```

Comment:
Should be `DatasetCore` rather than `Dataset`?

Reply:
Good point. Probably right!